src/device_discovery.c

```diff
@@ -0,0 +1,142 @@
+#include <libcaer/devices/device_discover.h>
+
+int main(void) {
+	caerDeviceDiscoveryResult discovered;
+	ssize_t result = caerDeviceDiscover(CAER_DEVICE_DISCOVER_ALL, &discovered);
+
+	if (result < 0) {
+		// Error!
+		return (EXIT_FAILURE);
+	}
+
+	printf("\n");
+	printf("Device discovery: found %zi devices.\n", result);
+	printf("\n");
+
```

Comment:
printf in C++ code, not great

Reply:
Deleted, but issue still persists in c++ equivalent (device_discovery.cpp). I will have a look into it to avoid as many print statements.